### PR TITLE
unbind: Add 'stack_or_none'

### PR DESCRIPTION
**Pull request by danthe3rd**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #518
* #511
* #510
* #509
* #505
* #508
* #504
* #491
* #502
* #490
* __->__ #498
* #488
* #487
* #486

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -184,7 +184,7 @@ run_coverage: &run_coverage
       when: always
       command: |
         source $BASH_ENV
-        CUDA_LAUNCH_BLOCKING=1 $CONDA_PYTHON -m pytest --junitxml=test-results/junit.xml --verbose --timeout 600 --cov-report=xml --cov=./ tests
+        CUDA_LAUNCH_BLOCKING=1 $CONDA_PYTHON -m pytest --junitxml=test-results/junit.xml --verbose --cov-report=xml --cov=./ tests
```
Comment (fmassa) [inline .circleci/config.yml:187]: Was this change necessary?

**Review by fmassa — APPROVED** [review]
Thanks!

**Comment by devonbrackbill** [comment]
I think this line causes a problem on installation: https://github.com/facebookresearch/xformers/commit/0428e12accab31cc3a5677b47b61ba8e7b6dfad2#r89809636
